app/domain/authentication/authn_k8s/k8s_host.rb

```diff
@@ -12,22 +13,27 @@
 
 module Authentication
   module AuthnK8s
+
+    Err ||= Errors::Authentication::AuthnK8s
+
     class K8sHost
       extend Forwardable
 
       attr_reader :account, :service_name, :csr
 
       def self.from_csr(account:, service_name:, csr:)
         cn = csr.common_name
-        raise ArgumentError, 'CSR must have a CN entry' unless cn
-
+        raise Err::CSRMissingCNEntry.new(csr.subject_to_s, csr.spiffe_id.to_s, account, service_name) unless cn
```

Comment:
when the line is too long we prefer to not have it multilined. this is better:
```
unless cn
  raise Err::CSRMissingCNEntry.new(csr.subject_to_s, csr.spiffe_id.to_s, account, service_name)
end
```

Reply:
got it

Reply:
This will work for CSRMissingCNEntry but not for CertMissingCNEntry as it accepts four arguments